File: Dddml.Wms.Services/Domain/InventoryItem/InventoryItemUtils.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Dddml.Wms.Domain.InventoryItem
{
    public static partial class InventoryItemUtils
    {

        public static void CreateOrUpdateInventoryItems(IInventoryItemApplicationService invItemApplicationService, IList<ICreateInventoryItemEntry> inventoryItemEntries)
        {
            foreach (var e in inventoryItemEntries)
            {
                var iitem = invItemApplicationService.Get(e.InventoryItemId);
                if (iitem == null)
                {
                    var createInventoryItem = new CreateInventoryItem();
                    createInventoryItem.InventoryItemId = e.InventoryItemId;
                    createInventoryItem.Entries.Add(e);
                    invItemApplicationService.When(createInventoryItem);
                }
                else
                {
                    var updateInventoryItem = new MergePatchInventoryItem();
                    updateInventoryItem.InventoryItemId = e.InventoryItemId;
                    updateInventoryItem.Version = iitem.Version; // /////////////////
                    updateInventoryItem.InventoryItemEntryCommands.Add(e);
                    invItemApplicationService.When(updateInventoryItem);
                }
            }
        }

    }
}

File: java/Dddml.Wms.JavaServices/src/main/java/org/dddml/wms/domain/inventoryitem/InventoryItemUtils.java
package org.dddml.wms.domain.inventoryitem;

/**
 * Created by yangjiefeng on 2018/1/31.
 */
public class InventoryItemUtils {

    public static void createOrUpdateInventoryItems(InventoryItemApplicationService invItemApplicationService, java.util.List<InventoryItemEntryCommand.CreateInventoryItemEntry> inventoryItemEntries) {
        for (InventoryItemEntryCommand.CreateInventoryItemEntry e : inventoryItemEntries) {
            InventoryItemState iitem = invItemApplicationService.get(e.getInventoryItemId());
            if (iitem == null) {
                InventoryItemCommand.CreateInventoryItem createInventoryItem = new AbstractInventoryItemCommand.SimpleCreateInventoryItem();
                createInventoryItem.setInventoryItemId(e.getInventoryItemId());
                createInventoryItem.getCreateInventoryItemEntryCommands().add(e);
                invItemApplicationService.when(createInventoryItem);
            } else {
                InventoryItemCommand.MergePatchInventoryItem updateInventoryItem = new AbstractInventoryItemCommand.SimpleMergePatchInventoryItem();
                updateInventoryItem.setInventoryItemId(e.getInventoryItemId());
                updateInventoryItem.setVersion(iitem.getVersion()); // /////////////////
                updateInventoryItem.getInventoryItemEntryCommands().add(e);
                invItemApplicationService.when(updateInventoryItem);
            }
        }
    }

}
